modules/checklist/assets/js/components/header.js

```diff
@@ -2,9 +2,37 @@ import { Typography, CloseButton, AppBar, Divider, Toolbar } from '@elementor/ui
 import { __ } from '@wordpress/i18n';
 import Progress from './progress';
 import PropTypes from 'prop-types';
+import { useState, useEffect } from 'react';
 
 const Header = ( { steps } ) => {
-	const closeChecklist = () => {
+	const [ closedForFirstTime, setClosedForFirstTime ] = useState( null );
+	const fetchStatus = async () => {
```

Comment:
Can you separate all the api calls in an `api` folder? Look at the notifications module.

Reply:
This will be done in a separate PR